src/metal/window-proxy.ts

```diff
@@ -26,97 +24,74 @@ interface WindowProxy {
   getWidth: Function;
   rAF: Function;
   meta: MetaInterface;
-  onWindowIsDirtyListeners: OnWindowIsDirtyInterface[];
-  __destroy__: Function;
+  version: number;
+  lastVersion: number;
+  updateMeta: Function;
+  isDirty: boolean;
 }
 
 const hasDOM = !!((typeof window !== 'undefined') && window && (typeof document !== 'undefined') && document);
 const hasRAF = hasDOM && !!window.requestAnimationFrame;
-const throttleDelay: number = 30;
 
-let resizeTimeout: number = 0;
-let scrollTimeout: number = 0;
 let W: WindowProxy = {
-  hasRAF,
   hasDOM,
+  hasRAF,
   getScrollTop: nop,
   getScrollLeft: nop,
   getHeight: nop,
   getWidth: nop,
-  onWindowIsDirtyListeners: [],
   rAF: hasRAF ? window.requestAnimationFrame.bind(window) : (callback: Function) => { callback(); },
   meta: {
     width: 0,
     height: 0,
     scrollTop: 0,
-    scrollLeft: 0
+    scrollLeft: 0,
+    x: 0,
+    y: 0,
+    top: 0,
+    left: 0
   },
-  __destroy__() {
-    this.onWindowIsDirtyListeners = [];
+  version: 0,
+  lastVersion: 0,
+  updateMeta: nop,
+  get isDirty(): boolean {
+    return W.version !== W.lastVersion;
   }
 };
 
+export function validateState() {
```

Comment:
name is confusing, I expected invalidate() or incrementVersion() or makeStateDirty()